### PR TITLE
Versions/2.0

### PR DESCRIPTION
**Pull request by WeilongYE**
Few changes
@jake5452 @ElizabethHollander

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <package>
   <name>cape_ros</name>
-  <version>1.0.0</version>
+  <version>1.0.3</version>
   <description>ROS wrapper for CAPE plane and cylinder extraction</description>
```
Comment (danielhailuA) [inline package.xml]: Since it is being used for only plane extraction, the description should reflect that.

File: src/cape_ros_node.cpp [hunk 2]
```diff
@@ -65,11 +65,11 @@ std::array<uint8_t, 3> hsvToRgb(const double h, const double s, const double v)
 
 CapeRosNode::CapeRosNode(ros::NodeHandle nh)
   : nh_(nh)
-  , cos_angle_max_(std::cos(M_PI / 30))
+  , cos_angle_max_(std::cos(M_PI / 45))
```
Comment (danielhailuA) [inline src/cape_ros_node.cpp]: Since the cos_angle_max_ is read from param why change it to M_PI/45 as default instead of M_PI/30?

File: src/cape_ros_node.cpp [hunk 3]
```diff
@@ -78,12 +78,17 @@ CapeRosNode::CapeRosNode(ros::NodeHandle nh)
 
   constexpr auto deg_to_rad = M_PI / 180;
 
-  auto max_angle = 6.0; // degrees
+  auto max_angle = 4.0; // degrees
```
Comment (danielhailuA) [inline src/cape_ros_node.cpp]: Is this default value or are you reading it from nh_.param ?

File: src/cape_ros_node.cpp [hunk 4]
```diff
@@ -65,11 +65,11 @@ std::array<uint8_t, 3> hsvToRgb(const double h, const double s, const double v)
 
 CapeRosNode::CapeRosNode(ros::NodeHandle nh)
   : nh_(nh)
-  , cos_angle_max_(std::cos(M_PI / 30))
+  , cos_angle_max_(std::cos(M_PI / 45))
   , max_merge_dist_(50.0)
   , depth_cutoff_(5.0)
   , cylinder_detection_(false)
```
Comment (danielhailuA) [inline src/cape_ros_node.cpp:71]: `cylinder_detection_` is always false. No extraction of cylinder is done in this code.

File: src/cape_ros_node.cpp [hunk 5]
```diff
@@ -78,12 +78,17 @@ CapeRosNode::CapeRosNode(ros::NodeHandle nh)
 
   constexpr auto deg_to_rad = M_PI / 180;
 
-  auto max_angle = 6.0; // degrees
+  auto max_angle = 4.0; // degrees
   nh_.param("max_angle", max_angle, max_angle);
   cos_angle_max_ = std::cos(max_angle * deg_to_rad);
-  nh_.param("max_merge_distance", max_merge_dist_, max_merge_dist_);
+  nh_.param("max_merge_dist", max_merge_dist_, max_merge_dist_);
```
Comment (danielhailuA) [inline src/cape_ros_node.cpp:84]: Is the parameter name "max_merge_distance" or "max_merge_dist" ? Why the change?

File: src/cape.cpp [hunk 6]
```diff
@@ -165,7 +164,7 @@ void Cape::process(Eigen::MatrixXf& cloud_array, cv::Mat& seg_out,
     new_ps.fitPlane();
 
     // 4. Model fitting
-    if (new_ps.score > 100) {
+    if (new_ps.score > 1) {
```
Comment (danielhailuA) [inline src/cape.cpp:167]: Is there reason why the previous value was high as 100  for Model fitting? Did it work before your changes?

File: package.xml (hunk 1 above)
Comment (WeilongYE) [reply package.xml]: We don't use the cylinder extraction but the code still has cylinder extraction, so I think it's OK to keep it.

File: src/cape_ros_node.cpp (hunk 2 above)
Comment (WeilongYE) [reply src/cape_ros_node.cpp]: I made the default value the same as in the parameters.

File: src/cape_ros_node.cpp (hunk 4 above)
Comment (WeilongYE) [reply src/cape_ros_node.cpp:71]: Yes, we are not using cylinder detection.

File: src/cape_ros_node.cpp (hunk 5 above)
Comment (WeilongYE) [reply src/cape_ros_node.cpp:84]: Just wanted to make it the same as max_merge_dist_

**Review by danielhailuA — APPROVED** [review]
Changes fine tune parameters and changes reflected based on tests and trials.

File: src/cape_ros_node.cpp [hunk 7]
```diff
@@ -219,19 +241,33 @@ void CapeRosNode::depthCallback(const sensor_msgs::ImagePtr& image)
...
 
-  cv_bridge::CvImageConstPtr image_ptr = cv_bridge::toCvShare(image, image->encoding);
+  cv_bridge::CvImageConstPtr image_ptr;
+  try
+  {
+    image_ptr = cv_bridge::toCvShare(image, image->encoding);
+  }
+  catch (const cv_bridge::Exception& e)
+  {
+    ROS_ERROR_STREAM("Could not convert depth image. " <<
```
Comment (ElizabethHollander) [inline src/cape_ros_node.cpp]: I'm noticing that some ROS_ERROR messages start with "Cape:" while others don't. I'm assuming they should all begin the same way for consistency.

File: src/cape_ros_node.cpp [hunk 8]
```diff
@@ -282,7 +322,19 @@ void CapeRosNode::depthCallback(const sensor_msgs::ImagePtr& image)
...
   {
     overlaySegmentsOnImage(intensity_image_ptr_->image, seg_output_);
-    image_overlay_pub_.publish(cv_bridge::CvImage(image->header, "rgb8", overlay_image_).toImageMsg());
+    sensor_msgs::ImagePtr image_msg;
+    try {
+      image_msg = cv_bridge::CvImage(image->header, "rgb8", overlay_image_).toImageMsg();
+    }
+    catch (const cv_bridge::Exception& e)
+    {
+      ROS_ERROR_STREAM("cape: Could not convert overlay_image_ to senor_msgs::Image. " <<
```
Comment (ElizabethHollander) [inline src/cape_ros_node.cpp]: Capitalize "cape"

File: src/cape_ros_node.cpp [hunk 9]
```diff
@@ -173,28 +184,39 @@ void CapeRosNode::overlaySegmentsOnImage(const cv::Mat& img, const cv::Mat& seg)
...
 {
-  cape_ros::PlanesPtr planes = boost::make_shared<cape_ros::Planes>();
+  planes_ = boost::make_shared<cape_ros::Planes>();
+
+  try {
+    planes_->segments = *cv_bridge::CvImage(header, "mono8", seg_output_).toImageMsg();
+  }
+  catch (const cv_bridge::Exception& e)
+  {
+    ROS_ERROR_STREAM("cape:Could not convert seg_output_ to planes_->segments. " <<
```
Comment (ElizabethHollander) [inline src/cape_ros_node.cpp]: capitalize "cape" and add space before the word "could"

File: src/cape_ros_node.cpp [hunk 10]
```diff
@@ -322,7 +374,21 @@ void CapeRosNode::reset()
...
-  intensity_image_ptr_ = cv_bridge::toCvShare(image, image->encoding);
+  if (intrinsics_ready_)
+  {
+    try
+    {
+      intensity_image_ptr_ = cv_bridge::toCvShare(image, image->encoding);
+    }
+    catch (const cv_bridge::Exception& e)
+    {
+      ROS_ERROR_STREAM("Could not convert intensity image. " <<
```
Comment (ElizabethHollander) [inline src/cape_ros_node.cpp]: Add "Cape:" to beginning of error message.

File: package.xml [hunk 11]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <package>
   <name>cape_ros</name>
-  <version>1.0.9</version>
+  <version>1.0.10</version>
```
Comment (danielhailuA) [inline package.xml]: What should the version number be or is it arbitrary ?

File: launch/cape_ros_node.launch [hunk 12]
```diff
@@ -1,10 +1,10 @@
 <launch>
-
+  <arg name="ns"    default=""/>
```
Comment (danielhailuA) [inline launch/cape_ros_node.launch:2]: Why is the default namespace an empty string?

**Review by ElizabethHollander — APPROVED** [review]

**Review by danielhailuA — APPROVED** [review]
Changes fine tune parameters and changes reflected based on tests and trials.

**Review by afakihcpr — APPROVED** [review]
